imap_processing/swe/l1a/swe_science.py

```diff
@@ -119,9 +119,7 @@ def swe_science(l0_dataset: xr.Dataset, data_version: str) -> xr.Dataset:
     # 4. Reshape the data to 180 x 7
     raw_science_array = np.array(
         [
-            np.frombuffer(
-                int(binary_string, 2).to_bytes(1260, byteorder="big"), dtype=np.uint8
-            ).reshape(180, 7)
+            np.frombuffer(binary_string, dtype=np.uint8).reshape(180, 7)
```

Comment:
Nice! This is what I was hoping for with the binary parameter types, being able to avoid the conversion to string and back again.

Reply:
It was nice for SWE but for some other instrument, unpacking scheme doesn't line up good with bytes. That's why I had to convert bytes to binary string for those.